core/src/main/java/org/htsjdk/core/utils/PathSpecifier.java

```
@@ -0,0 +1,238 @@
+        for (FileSystemProvider provider: FileSystemProvider.installedProviders()) {
+            if (provider.getScheme().equalsIgnoreCase(uri.getScheme())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public boolean isPath() {
+        try {
+            return getCachedPath() != null || toPath() != null;
+        } catch (ProviderNotFoundException |
+                FileSystemNotFoundException |
+                IllegalArgumentException |
+                HtsjdkException |
+                AssertionError e) {
+            // jimfs throws an AssertionError that wraps a URISyntaxException when trying to create path where
+            // the scheme-specific part is missing or incorrect
+            pathFailureReason = e.getMessage();
+            return false;
+        }
+    }
+
+    @Override
+    public URI getURI() {
+        return uri;
+    }
+
+    @Override
+    public String getURIString() {
+        return getURI().toString();
+    }
+
+    /**
+     * Return the raw input string as provided to the constructor.
+     */
+    @Override
+    public String getRawInputString() { return rawInputString; }
+
+    /**
+     * Resolve the URI to a {@link Path} object.
+     *
+     * @return the resulting {@code Path}
+     * @throws HtsjdkException if an I/O error occurs when creating the file system
+     */
+    @Override
+    public Path toPath() {
+        if (getCachedPath() != null) {
+            return getCachedPath();
+        } else {
+            final Path tmpPath = Paths.get(getURI());
+            setCachedPath(tmpPath);
+            return tmpPath;
+        }
+    }
+
+    @Override
+    public String getToPathFailureReason() {
+        if (pathFailureReason == null) {
+            try {
+                toPath();
+                return String.format("'%s' appears to be a valid Path", rawInputString);
+            } catch (ProviderNotFoundException e) {
```

This can be summarize better as:
```suggestion
            } catch (final Exception e) {
                return String.format("%s: %s", e.getClass().getSimpleName(), e.getMessage());
            }
```

---

Only in case the exception is not returned instead, which looks much better.

---

I generally try to avoid catching `Exception`, especially without re-throwing it, since it casts too wide of a net and can mask underlying problems.